File: 백준/Bronze/2884. 알람 시계/알람 시계.py
h, m = map(int, input().split())

a = m - 45

if(a < 0):
    if(h == 0) :
        h = 23
        m = 60 + a
    else:
        h = h - 1
        m = 60 + a
else:
    m = a
    
print(h, m)

File: 백준/Bronze/2884. 알람 시계/알람 시계.kt
fun main() {
    var (a, b) = readln().split(" ").map { it.toInt() }

    if (b - 45 < 0) {
        if (a == 0) {
            a = 23
        } else {
            a -= 1
        }
        b += 60 - 45
    } else {
        b -= 45
    }
    println("$a $b")
}